src/rpc.rs

```diff
@@ -367,3 +369,55 @@ pub fn get_id_specific_size(num: Block) -> HashMap<u32, u32> {
 	}
 	index
 }
+//parsing the urls given in the vector of urls
+pub fn parse_urls(urls: Vec<String>) -> Result<Vec<url::Url>> {
+	urls.iter()
+		.map(|url| url::Url::parse(url))
+		.map(|r| r.map_err(|parse_error| anyhow!("Cannot parse URL: {}", parse_error)))
+		.collect::<Result<Vec<_>>>()
+}
+
+//fn to check the ws url is working properly and return it
+pub async fn check_connection(
```

Comment:
Are we checking the entire provided list of wss if they are connectable?

Reply:
yes we are! if one connection is successfull then we return that and if the same is lost then we again try to connect to other one!